application/models/watch.go

```diff
@@ -147,30 +148,126 @@ func (w *Watch) Destroy() error {
 	return DB.Destroy(w)
 }
 
-// matchesPost returns true if the Watch's Location is near the Post's Destination
+func (w *Watch) Meeting(ctx context.Context) (*Meeting, error) {
+	if w == nil || !w.MeetingID.Valid || CurrentUser(ctx).ID != w.OwnerID {
+		return nil, nil
+	}
+	meeting := &Meeting{}
+	if err := DB.Find(meeting, w.MeetingID); err != nil {
+		return nil, err
+	}
+	return meeting, nil
+}
+
+// matchesPost returns true if all non-null watch criteria match the post
 func (w *Watch) matchesPost(post Post) bool {
-	dest, err := post.GetDestination()
+	if w == nil {
+		domain.ErrLogger.Printf("nil receiver in Watch.matchesPost")
+		return false
+	}
+	compareFunctions := []func(*Watch, Post) bool{
```

Comment:
Cool.  Where did you find out how to do this - with struct methods as a slice of functions?

Reply:
https://stackoverflow.com/questions/31561369/how-do-i-declare-a-function-pointer-to-a-method-in-go